experiments/ClimaEarth/user_io/arg_parsing.jl

```diff
@@ -1,5 +1,21 @@
 import YAML
 
+abstract type AbstractModeType end
+abstract type AbstractSlabPlanetModeType <: AbstractModeType end
+abstract type AbstractSlabPlanetModeSubType1 <: AbstractSlabPlanetModeType end
+struct AMIP_mode <: AbstractModeType end
```

Comment:
Julia convention is to use camel case for struct names, so this would be `AMIPMode` (same for the structs below, e.g. `SlabplanetMode`